File: Python/OJ/20_04/fib_dp.py
class Solution:

    def fib(self, N: int) -> int:
        dp = [0, 1]+[0 for i in range(30)]
        if N <= 1:
            return 1
        else:
            for i in range(2,N):
                dp[i]+= dp[i-1]+dp[i-2]
        return dp[N-1]

File: C/OJ/Leetcode/2020-4/fibDP.c
int recur(long *dp, int N)
{
    if (N == 1)
    {
        return 0;
    }
    if (N == 2)
    {
        return 1;
    }
}

int fib(int N)
{
    static long dp[30] = {0};
    dp[1] = 1;
    if (N == 1)
    {
        return 0;
    }
    if (N == 2)
    {
        return 1;
    }
    return recur(N - 1) + recur(N - 2);
}
